### DISCUSSION TITLE
docusaurus cms

### DISCUSSION THREAD

**Discussion by PatelN123** [directus/directus#11648] [Archive]
Hi, 

Can this be used as a cms for docusaurus?

**Comment by xu4wang** [comment]
Theoretically yes.  The app can be used as the CMS UI. When a record is added/updated, a hook in server side can generate the markdown file required by docusaurus.

However it doesn't make much sense, since the database behind directus is wasted.

File based CMS is more suitable for docusaurus.

**Reply by rijkvanzanten** [reply]
What do you mean by

> However it doesn't make much sense, since the database behind directus is wasted.

?

**Reply by xu4wang** [reply]
I meant the power of the database, for example query/filter etc, is not used,  in my previous proposal.

**Comment by xu4wang** [comment]
Another way for the CMS scenario.

1. Create collections for the "docs, blogs, i18n,... etc" to hold the information needed for the docusaurus site.
2. Use directus APP to manage the content/database.
3. Whenever user want to publish content, a script can be trigger to 
   - retrieve information from directus database and generate markdown files for docusaurus. 
   - run docusaurus to generate static site

**Reply by PatelN123** [reply]
Thank you. 

Would I be able to get a widget or icon that shows on the top left of my docusaurus site that's shows if the user is logged in or not? - this is a key feature for me.

**Reply by xu4wang** [reply]
Not possible using docusaurus standard.  You may need to check docusaurus plugin or change docusaurus  implementation.

Just checked my doc site(also use docusaurus), the logo on upper left is set by docusaurus.config.js, and deployed as a static asset.

**Reply by PatelN123** [reply]
It is possible to `swizzle` the Navbar component, if I was to swizzle it, I can add custom elements. 

Would it be possible to add it then?

**Reply by rijkvanzanten** [reply]
> Would I be able to get a widget or icon that shows on the top left of my docusaurus site that's shows if the user is logged in or not? - this is a key feature for me.

It sounds like you're building a more dynamic site than most static site generators are built to handle 🤔 You might want to look into a static / dynamic hybrid like gatsby or gridsome to make life a little easier
